src/dsa/rpo_falcon512/keys.rs

```diff
@@ -41,69 +48,50 @@ impl From<PublicKey> for Word {
     }
 }
 
-// KEY PAIR
+// SECRET KEY
 // ================================================================================================
 
-/// A key pair (public and secret keys) for signing messages.
-///
-/// The secret key is a byte array of length [PK_LEN].
-/// The public key is a byte array of length [SK_LEN].
-#[derive(Debug, Clone, Copy, PartialEq)]
-pub struct KeyPair {
-    public_key: PublicKeyBytes,
-    secret_key: SecretKeyBytes,
+/// TODO: ADD DOCS
+#[derive(Debug, Clone)]
+pub struct SecretKey {
+    secret_key: B0,
+    tree: LdlTree,
 }
 
 #[allow(clippy::new_without_default)]
-impl KeyPair {
+impl SecretKey {
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
 
-    /// Generates a (public_key, secret_key) key pair from OS-provided randomness.
-    ///
-    /// # Errors
-    /// Returns an error if key generation fails.
-    #[cfg(feature = "std")]
-    pub fn new() -> Result<Self, FalconError> {
-        let mut public_key = [0u8; PK_LEN];
-        let mut secret_key = [0u8; SK_LEN];
-
-        let res = unsafe {
-            ffi::PQCLEAN_FALCON512_CLEAN_crypto_sign_keypair_rpo(
-                public_key.as_mut_ptr(),
-                secret_key.as_mut_ptr(),
-            )
-        };
+    /// Generates a secret key from OS-provided randomness.
+    pub fn new() -> Self {
```

Comment:
This should probably be available only in `std` context.

Reply:
Agreed and changed. This also intersects with the above comment